### PR TITLE
Problem 394: ScheduleTask was replaced by RunTask.

### PR DESCRIPTION
**Pull request by kegor**
**Problem: https://github.com/Starcounter/UniformDocs/issues/394**
```
System.MissingMethodException: Method not found: 'Void
Starcounter.Session.ScheduleTask(System.String, SessionTask, Boolean)'.    at 
Joozek78.Star.Async.Internal.StarcounterSessionSynchronizationContext.Post(SendOrPostCallback d, Object state)    at System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, Object state, Task& currentTask)
--
```

**Solution:**
Replace using `Session.ScheduleTask` on the `Session.RunAll`.

### CODE REVIEW

**Comment by kegor** [comment]
Few additional points regarding this task:
1. I checked the changes locally on `UniformDocs`, looks good (as usual):
progress_bar
2. As the part of this task i would change the first link above progress-bar to our newly forked project.
3. And main point: we need configured CI for this project to make possible make deploy on the NuGet. I didn't find existing in our TC using search, @iqbalskd maybe you could also try to find? If you'll don't find could you please setup this project as it was made for example here https://teamcity.starcounter.org/viewType.html?buildTypeId=Apps_StarcounterAuthorization_StarcounterAuthorizationNugetPublishMaster3x (Starcounter.Authorization nuget publish master (3.x)), additionally we probably need regular TC project for the `https://github.com/Starcounter/starcounter-async` `master` branch.

**Comment by warpech** [comment]
1. Great!
2. OK, please make that in the PR in KitchenSink where you will use this new fork
3. I don't think you *need* CI to upload a package to NuGet... These libraries: https://www.nuget.org/packages/Starcounter.Startup/, https://www.nuget.org/packages/Starcounter.Linq/ are manually released to NuGet without any job on TC

Btw. when you release the first version of this package in our namespace, please use a continuation of previous numbering (choose a version higher than `2.0.0`) and paste a link to the original package in the release notes.

**Review by warpech — CHANGES_REQUESTED** [review]
Please change the namespace `Joozek78.Star` to something more relevant that matches the style of namespaces used in our other NuGet packages.

**Comment by kegor** [comment]
@warpech, namespaces were updated.

[Update]: also maybe we can rename current repo to `Starcounter.Async` as well?

**Comment by warpech** [comment]
> [Update]: also maybe we can rename current repo to Starcounter.Async as well?

Please do. This will be aligned with other library repos.

Technically, I think it is better if @PatrykSzwer makes a review here instead of me.

**Review by warpech — APPROVED** [review]

File: Starcounter.Async/Internal/StarcounterSessionSynchronizationContext.cs [hunk 1]
```diff
@@ -25,13 +23,14 @@ public StarcounterSessionSynchronizationContext()
...
-            Session.ScheduleTask(SessionId,
-                (Session session, string sessionId) => {
+            Session.RunTask(SessionId,
+                (Session session, string sessionId) =>
+                {
                     if (session != null)
                     {
                         d(state);
                     }
-                }, waitForCompletion: true);
```
Comment (PatrykSzwer) [inline Starcounter.Async/Internal/StarcounterSessionSynchronizationContext.cs:34]: I see you removed `waitForCompletion` argument. It doesn't exist in `Session.RunTask` or its not needed anymore?

**Review by PatrykSzwer — APPROVED** [review]
I checked the code and build the lib locally. Except this one comment/question everything looks fine.

File: Starcounter.Async/Internal/StarcounterSessionSynchronizationContext.cs (hunk 1 above)
Comment (kegor) [reply Starcounter.Async/Internal/StarcounterSessionSynchronizationContext.cs:34]: Yeah, it doesn't exists in `Session.RunTask`.

**Pull request merged**